### PR TITLE
Task/cdd 2098 integrate headline manager to tables endpoint

### PR DESCRIPTION
**Pull request by phill-stanley**
# Description

This PR includes the the integration of the `CoreHeadline` manager in the tables endpoint to enable tabular data for headline charts.

Fixes #CDD-2098

---

## Type of change

Please select the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Tech debt item (this is focused solely on addressing any relevant technical debt)

---

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests at the right levels to prove my change is effective
- [ ] I have added screenshots or screen grabs where appropriate
- [x] I have added docstrings in the correct style [(google)](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings)

### CODE REVIEW

File: metrics/domain/tables/generation.py [hunk 1]
```diff
@@ -153,25 +160,77 @@ def create_multi_plot_output(self) -> list[dict[str, str | list[dict]]]:
...
+
+        Returns:
+            A list of dictionaries representing a plot and its values
+            Eg. [{label: "Amount", value: "55", IN_REPORTING_DELAY_PERIOD: False}]
+        """
+        return [
+            {
+                "label": "Amount",
+                "value": plot_values.get(plot_label),
+                IN_REPORTING_DELAY_PERIOD: False,
```
Comment (A-Ashiq) [inline metrics/domain/tables/generation.py:232]: Do we have reporting delay period for headline data?

File: metrics/interfaces/charts/access.py [hunk 2]
```diff
@@ -66,18 +66,18 @@ def _set_core_model_manager(self) -> Manager:
 
         Notes:
             The charts interface can be used to generate charts for
-            either `CoreTimerseries` or `CoreHeadline` data.
+            either `CoreTimerSeries` or `CoreHeadline` data.
```
Comment (A-Ashiq) [inline metrics/interfaces/charts/access.py:69]:
```suggestion
            either `CoreTimeSeries` or `CoreHeadline` data.
```

File: tests/unit/metrics/interfaces/tables/test_access.py [hunk 3]
```diff
@@ -28,31 +30,64 @@ def _setup_fake_time_series_for_plot(chart_plot_parameters: PlotParameters):
...
         When an instance of the `TablesInterface` is created
             without explicitly providing a `PlotsInterface`
         Then an instance of the `PlotsInterface` is created with the correct args
         """
         # Given
-        mocked_plots_collection = mock.MagicMock()
+        fake_plots_collection.plots[0].chart_type = (
+            ChartTypes.line_with_shaded_section.value
+        )
+        # mocked_plots_collection = mock.MagicMock()
```
Comment (A-Ashiq) [inline tests/unit/metrics/interfaces/tables/test_access.py:47]: Do you need this?

Comment (phill-stanley) [reply tests/unit/metrics/interfaces/tables/test_access.py]: Good catch, no this is not needed.

File: metrics/domain/tables/generation.py [hunk 4]
```diff
@@ -153,25 +160,77 @@ def create_multi_plot_output(self) -> list[dict[str, str | list[dict]]]:
...
+            For headline tabular data the label for each plot
+            is set to `Amount` to keep all values in the same
+
+        Returns:
+            A list of dictionaries representing a plot and its values
+            Eg. [{label: "Amount", value: "55", IN_REPORTING_DELAY_PERIOD: False}]
+        """
+        return [
+            {
+                "label": "Amount",
```
Comment (A-Ashiq) [inline metrics/domain/tables/generation.py:230]: Why do we have to fix the `label`?

File: metrics/interfaces/tables/access.py [hunk 5]
```diff
@@ -1,28 +1,56 @@
...
+        )
+        self.core_model_manager = core_model_manager or self._set_core_model_manager()
 
         self.plots_interface = plots_interface or PlotsInterface(
             plots_collection=self.plots_collection,
-            core_model_manager=core_time_series_manager,
+            core_model_manager=self.core_model_manager,
         )
 
+    def _set_core_model_manager(self) -> Manager:
```
Comment (A-Ashiq) [inline metrics/interfaces/tables/access.py:36]: Nitpick: `set` in the name feels misleading. When I read that I think, oh we're setting the corresponding model manager on the internal state of this class. But really we're just correct getting the model manager. Nothing is being persisted anywhere

Comment (phill-stanley) [reply metrics/interfaces/tables/access.py:36]: good point, I'll update the naming here.

File: metrics/domain/tables/generation.py (hunk 4 above)
Comment (phill-stanley) [reply metrics/domain/tables/generation.py:230]: The label for tabular data is used to create either multiple sections or columns depending on the structure of the payload. With time series the plot label is used to give a column/section heading and then the timeseries data is presented under that heading. So each section has a number of values against a date.

With headline data there are no dates, the plot is a single value, to present this data we want to combine the results into a single column and this is done by setting the label of each plot to the same value.

Comment (A-Ashiq) [reply metrics/domain/tables/generation.py:230]: Thanks for the explanation

**Review by A-Ashiq — APPROVED** [review]

**Pull request merged**